pkgs/development/tools/documentation/adrdox/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchFromGitHub, dmd }:
+
+stdenv.mkDerivation rec {
+  pname = "adrdox";
+  version = "v1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "adamdruppe";
+    repo = "adrdox";
+    rev = version;
+    sha256 = "1w0y158ab36385jhg6q9xjlb8cilfwimlh43q86n40l10vhn4bk9";
+  };
+
+  buildInputs = [ dmd ];
+
+  # We need a custom install phase because the makefile lacks an install
+  # target. If a future version of adrdox gets an install target, the custom
+  # install phase may be removed.
+  installPhase = ''
+    install -d $out/bin
+
+    install doc2 $out/bin
+
+    # doc2 expects the asset files to be in the same directory as the binary.
+    install skeleton-default.html script.js search-docs.js style.css $out/bin
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Documentation generator for D";
+    homepage = "https://github.com/adamdruppe/adrdox";
+    license = licenses.boost;
+    maintainers = [ maintainers.chloekek ];
+
+    # The Makefile passes -m64 to DMD, which causes it to generate a 64-bit
+    # executable. This hints at doc2 not working on other platforms?
+    platforms = platforms.x86_64;
```

Comment:
Might be worth checking if this is actually this is actually the case though tbh I'm not sure how to check that

Reply:
It also seems to work when a 32-bit binary is generated. I checked that by adding the following code:

```nix
configurePhase = ''
  sed -i 's/-m32//g' Makefile
'';
```

I could add a patch file that removes -m64 from the Makefile; in that case DMD will generate 32-bit or 64-bit code depending on the platform.

Reply:
I don't see the harm in doing that If it does not change the x64 version

Reply:
I think it might be better to use preConfigure instead of overriding the configure phase

Reply:
I tried to build a 32-bit binary like this:

```nix
{
  configurePhase = ''
    sed -i 's/-m64/-m32/g' Makefile
  '';
}
```

But that fails with a linker error because Nixpkgs only provides 64-bit libraries for it to link against, since I'm on a 64-bit computer.

So testing it probably requires 32-bit Nixpkgs or a 32-bit computer.

Reply:
Maybe if in doubt then don't change anything

Reply:
I think I don't know enough to know what the best way forward is

Reply:
I added a Makefile patch that removes the 64-bit restriction.